IDE/QNX/example-server/server-tls.c

```diff
@@ -0,0 +1,392 @@
+                        * potential padding */
+    word32 blackKeySz = 48;
+
+
+    /* The DER buffer for test case does not contain a black key, here we will
+     * extract the key information, cover it (make it a black key), and then
+     * import that into 'keyOut' which will create a secure partition on import.
+     *
+     * These steps of covering and importing can be avoided with a call to
+     * wc_ecc_make_key which creates a new black key that never leaves the CAAM.
+     * The cover process is only being used to make use of an existing key that
+     * matches the test certificate.
+     */
+    wc_ecc_init(&notSecure);
+    if (wc_EccPrivateKeyDecode(der, &idx, &notSecure, derSz) != 0) {
+        printf("Error extracting ECC der buffer\n");
+        ret = -1;
+        goto done;
+    }
+
+    /* Get the uncovered key values */
+    if (wc_ecc_export_private_raw(&notSecure, NULL, NULL, NULL, NULL, d, &dSz)
+            != 0) {
+        printf("Error getting ecc key values\n");
+        ret = -1;
+        goto done;
+    }
+
+    if (wc_ecc_export_x963(&notSecure, x963, &x963Sz) != 0) {
+        printf("Error getting ecc public key\n");
+        ret = -1;
+        goto done;
+    }
+
+    /* Cover up the private key (make it a black key, encrypting with AES-CCM)*/
+    XMEMSET(blackKey, 0, blackKeySz);
+    if (wc_caamCoverKey(d, dSz, blackKey, &blackKeySz, 0) != 0) {
+        printf("Error covering up the private key\n");
+        ret = -1;
+        goto done;
+    }
+
+    /* Sanity check to compare covered key versus plain text */
+    if (XMEMCMP(d, blackKey, dSz) == 0) {
+        printf("Something went wrong with key cover!!\n");
+        ret = -1;
+        goto done;
+    }
+
+    if (test_blob(blackKey, blackKeySz - 16) != 0) {/*-16 byte for MAC padding*/
+        printf("test blob failed\n");
+        ret = -1;
+        goto done;
+    }
+
+    /* zero out the plain text buffer and free key */
+    XMEMSET(d, 0, dSz);
+
+    /* Import the black key into a ecc_key structure (this step tries to create
+     * a partition of secure memory marked CPS)
+     */
+    ret = wc_ecc_import_private_key(blackKey, blackKeySz, x963, x963Sz, keyOut);
+    if (ret != 0) {
+        printf("Error importing black key into secure memory %d\n", ret);
+        wc_ecc_free(keyOut);
+        goto done;
+    }
+    printf("blackKeySz = %d, virtual secure address ecc_key.blackKey = 0x%08X\n",
+            blackKeySz, keyOut->blackKey);
+
+    ret = 0;
+done:
+
+    wc_ecc_free(&notSecure);
+    return ret;
+}
+
+
+/* callback function for creating ECC signatures */
+static int TLS_ECC_Sign_callback(WOLFSSL* ssl, const unsigned char* in,
+        unsigned int inSz, unsigned char* out, word32* outSz,
+        const unsigned char* keyDer, unsigned int keySz,
+        void* ctx)
+{
+    ecc_key* blackKey;
+
+    printf("Using ECC sign callback\n");
+
+    if (ctx == NULL) {
+        printf("Was expecting a black key passed along with WOLFSSH\n");
```

Comment:
This says `WOLFSSH`?

Reply:
typo from switching between wolfSSH dev over to wolfSSL